src/control-plane/backend/lambda/listen-state-status/index.ts

```
@@ -0,0 +1,300 @@
+import { ExecutionStatus } from '@aws-sdk/client-sfn';
+import { DeleteTopicCommand, ListSubscriptionsByTopicCommand, SNSClient, UnsubscribeCommand } from '@aws-sdk/client-sns';
+import { DynamoDBDocumentClient, UpdateCommand, QueryCommandInput, paginateQuery, UpdateCommandInput, ScanCommandInput, paginateScan } from '@aws-sdk/lib-dynamodb';
+import { NativeAttributeValue } from '@aws-sdk/util-dynamodb';
+import { EventBridgeEvent } from 'aws-lambda';
+import { ExecutionDetail, PipelineStatusType } from '../../../../common/model';
+import { logger } from '../../../../common/powertools';
+import { aws_sdk_client_common_config, marshallOptions, unmarshallOptions } from '../../../../common/sdk-client-config';
+import { CFN_RULE_PREFIX, CFN_TOPIC_PREFIX } from '../api/common/constants';
+
+const ddbClient = new DynamoDBClient({
+  ...aws_sdk_client_common_config,
+});
+
+const docClient = DynamoDBDocumentClient.from(ddbClient, {
+  marshallOptions: { ...marshallOptions },
+  unmarshallOptions: { ...unmarshallOptions },
+});
+
+const clickStreamTableName = process.env.CLICKSTREAM_TABLE_NAME ?? '';
+const prefixTimeGSIName = process.env.PREFIX_TIME_GSI_NAME ?? '';
+
+export interface StepFunctionsExecutionStatusChangeNotificationEventDetail extends ExecutionDetail {
+  startDate?: number;
+  stopDate?: number;
+  input?: any;
+  output?: any;
+}
+
+export const handler = async (
+  event: EventBridgeEvent<'Step Functions Execution Status Change', StepFunctionsExecutionStatusChangeNotificationEventDetail>): Promise<void> => {
+
+  const eventDetail = event.detail;
+  const executionName = eventDetail.name;
+  if (!executionName?.startsWith('main-')) {
+    return;
+  }
+  logger.info('Detail: ', { executionName: eventDetail.name, status: eventDetail.status });
+
+  const pipelineId = executionName.split('-')[1];
+  const pipeline = await getPipeline(pipelineId);
+  if (!pipeline) {
+    logger.error('Failed to get pipeline by pipelineId: ', { pipelineId });
+    return;
```

This swallows this event. I prefer to throw the exception to retry it later.